commands/project.go

```diff
@@ -0,0 +1,93 @@
+package commands
+
+//handle the project logic for rhmap.
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/feedhenry/rhm/storage"
+	"github.com/urfave/cli"
+)
+
+//ProjectCmd constructs the required writer in order to send the response to the right place.
+type projectCmd struct {
+	out      io.Writer
+	in       io.Reader
+	response func(*http.Request) (*http.Response, error)
+	//response   http.Client
+	newrequest func(string, string, io.Reader) (*http.Request, error)
+	store      storage.Storer
+}
+
+//Project Defines our cli command including its flags and usage then returns the command to allow a user to do specific operations on projects
+func (pc *projectCmd) Project() cli.Command {
+	return cli.Command{
+		Name:        "project",
+		Action:      pc.projectAction,
+		Usage:       "project",
+		Description: "project will allow a user to create,update,delete and list projects in rhm",
+	}
+}
+
+//this is the data structure for posting to the server
+type projectParams struct {
+	Domain string `json:"d"`
+}
+
+//projectAction is where the logic is pulled together to perform the command. This funtion conforms to the cli action
+func (pc *projectCmd) projectAction(ctx *cli.Context) error {
+	var (
+		url = "%s/box/api/projects"
+	)
+	userData, err := pc.store.ReadUserData()
+	if err != nil {
+		return cli.NewExitError("could not read userData "+err.Error(), 1)
+	}
+	fullURL := fmt.Sprintf(url, userData.Host)
+	newrequest, err := pc.newrequest("GET", fullURL, nil)
+	if err != nil {
+		return cli.NewExitError("could not create new request object "+err.Error(), 1)
+	}
+
+	// create a cookie
+	expiration := time.Now().Add(365 * 24 * time.Hour)
```

Comment:
Is the expiration actually required here? We are just sending it rather than setting it